Extensions/BitmapText/bitmaptextruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,209 @@
+        this._pixiObject.fontName = fontName;
+        this.updatePosition();
+      })
+      .catch((error) => {
+        console.error('Error while loading a bitmapFont resource:', error);
+      });
+  } else {
+    this.updateFont();
+    this.updateFontSize();
+  }
+
+  runtimeScene
+    .getLayer('')
+    .getRenderer()
+    .addRendererObject(this._pixiObject, runtimeObject.getZOrder());
+
+  // Set the anchor in the center, so that the object rotates around
+  // its center.
+  this._pixiObject.anchor.x = 0.5;
+  this._pixiObject.anchor.y = 0.5;
+
+  this.updateAlignment();
+  this.updateTextContent();
+  this.updatePosition();
+  this.updateAngle();
+  this.updateOpacity();
+  this.updateWrappingWidth();
+};
+
+gdjs.BitmapTextRuntimeObjectRenderer = gdjs.BitmapTextRuntimeObjectPixiRenderer;
+
+gdjs.BitmapTextRuntimeObjectPixiRenderer.prototype.getRendererObject = function () {
+  return this._pixiObject;
+};
+
+gdjs.BitmapTextRuntimeObjectPixiRenderer.prototype.onDestroy = function () {
+  // Mark the font from the object not used anymore.
+  gdjs.BitmapFontManager.removeFontUsed(this._pixiObject._fontName);
+};
+
+gdjs.BitmapTextRuntimeObjectPixiRenderer.prototype._ensureFontAvailableAndGetFontName = function (
+  oldFont
+) {
+  const slugFontName =
+    this._bitmapFontStyle.fontFamily +
+    '-' +
+    this._bitmapFontStyle.fontSize +
+    '-' +
+    this._bitmapFontStyle.fill +
+    '-bitmapFont';
+
+  // Load the font if it's not available yet.
+  if (!PIXI.BitmapFont.available[slugFontName]) {
+    console.info('Generating font "' + slugFontName + '" for BitmapText.');
+    PIXI.BitmapFont.from(slugFontName, this._bitmapFontStyle, {
+      chars: [
+        [' ', '~'], // All the printable ASCII characters
+      ],
+    });
+  }
+
+  gdjs.BitmapFontManager.setFontUsed(slugFontName);
+
+  if (oldFont) {
+    gdjs.BitmapFontManager.removeFontUsed(oldFont);
+  }
+
+  // TODO: find a way to unload the BitmapFont that are not used anymore, otherwise
+  // we risk filling up the memory with useless BitmapFont - especially when the user
+  // plays with the color/size.
+
+  return slugFontName;
+};
+
+gdjs.BitmapTextRuntimeObjectPixiRenderer.prototype.updateFont = function () {
+  let bitmapFontResourceName = this._object._bitmapFontFile;
+  let bitmapTextureResourceName = this._object._bitmapTextureFile;
+
+  // Get the texture used in the bitmap font
+  let texture = this._object._runtimeScene
+    .getGame()
+    .getImageManager()
+    .getPIXITexture(bitmapTextureResourceName);
+
+  // Get the bitmap font file and use the texture for generate the bitmapFont (PIXI.BitmapFont) and return the fontName ready to use.
+  this._object._runtimeScene
+    .getGame()
+    .getBitmapFontManager()
+    .loadBitmapFont(bitmapFontResourceName, texture)
+    .then((fontName) => {
+      this._pixiObject.fontName = fontName;
+    })
+    .catch((error) => {
+      console.error('Error while loading a bitmapFont resource:', error);
+    });
```

Comment:
See my other explanation, it's possible that you can remove the promise here, because by preloading font data, you can call `...getBitmapFontManager.getBitmapFont(bitmapFontResourceName, texture)`.

Reply:
Also just to make things clear:

- In the IDE, we're fine having resources loaded "on the go". This means that in the "InstanceRenderer" of the IDE, we check if the bitmap font resource name (or the texture name) changed compared to what we display. If yes, we kick off the loading of the bitmap font resource name and then we call BitmapFont.install.
  - This means that for a few (milli)seconds, the old font is shown.
  
- In the runtime, we preload everything (at least for now!). This means that your BitmapFontManager should be able to give you instantly the font data (if it exists, otherwise the resource name is probably wrong => you can stop). So we should be able to avoid a promise :) 
  - I'm still unsure if in the future we'll rework this to have promises and support objects dynamically loading a resource... but for now I think we can do by preloading the bitmap font data. Note that the `BitmapFont.install` is only done at the last time!
  
For this last point, we could argue "can we do BitmapFont.install sooner, like during the preloading?". The problem is that the xml/fnt data file does NOT have the texture... so we can't! But maybe later we'll create resources that are "embedding other resources".